src/FilterService/ListHelper.php

```diff
@@ -67,18 +68,22 @@ public function setupProductList(
             $params['currentPage'] = (int)$params['page'];
             $offset = $pageLimit * ($params['page'] - 1);
         }
+
+        $productList->setLimit((int)$pageLimit);
+
         if ($filterDefinition->getAjaxReload()) {
             if ($loadFullPage && !$excludeLimitOfFirstpage) {
                 $productList->setLimit($pageLimit);
-            } elseif ($loadFullPage && $excludeLimitOfFirstpage) {
+            }
+            if ($loadFullPage && $excludeLimitOfFirstpage) {
```

Comment:
this part
https://github.com/pimcore/ecommerce-framework-bundle/blob/e6ca0e41fc37f22ffafec47660ea3faca975d66b/src/FilterService/ListHelper.php#L75-L84
could be refactored as 

```
            $productList->setLimit($limitOnFirstLoad);
            if ($loadFullPage) {
                $productList->setLimit($pageLimit);
                if ($excludeLimitOfFirstpage){
                    $offset += $limitOnFirstLoad;
                    $productList->setLimit($pageLimit - $limitOnFirstLoad);
                }
            }
```